# HEALTHCARE INSURANCE COST ANALYSIS

## Objectives

* Using the right packages and tools to clean and transform a Healthcare Insurance Dataset to further draw insights from the data.Such packages include: Feature engine, scikit learn and pandas.
* Using the right packages to visualize the data for better analysis. Such packages include: Matplotlib, Seaborn and Plotly.
* Using the information from the analysis to provide data-driven hypothesis for better decision making.

## Inputs

* The dataset: 'insurance.csv'

## Outputs

* Write here which files, code or artefacts you generate by the end of the notebook 





---

# Change working directory

* We are assuming you will store the notebooks in a subfolder, therefore when running the notebook in the editor, you will need to change the working directory

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [1]:
import os
current_dir = os.getcwd()
current_dir

'/home/alban/Project 1/Project1/jupyter_notebooks'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chir() defines the new current directory

In [ ]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

# For Google Colab compatibility
import sys
if 'google.colab' in sys.modules:
    print("Running in Google Colab")
else:
    print("Running locally")

You set a new current directory


Confirm the new current directory

In [11]:
current_dir = os.getcwd()
current_dir

'/home/alban'

# Section 1

Section 1 content

## Importing the packages or dependencies for the analysis


In [1]:
import pandas as pd
import numpy as np
import feature_engine 
import matplotlib.pyplot as plt
import seaborn as sns
import sklearn

# Data Extraction
* We load the dataset 'insurance.csv' from the Raw Data subfolder using the pandas method '.read_csv()'. 

In [6]:
df=pd.read_csv('/mnt/chromeos/MyFiles/Downloads/archive.zip')
df.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

# Data Transformation
* This involves cleaning the data by checking for inadequacies such as missing values, duplicates and outliers. Before that we can check for the general overview of the dataset using the '.info()' method.

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


* The info method showed that the dataset contains 7 columns (4 numericals and 3 categorical), it also tells us the number of rows (1338) present in the dataset alongside the different data types (float, integer and object) of the variables. Next, we check for missing values in the dataset using the '.isnull().sum()' method.

In [10]:
df.isnull().sum()

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64

* The above method showed zero cases of missing values, therefore, we can go ahead and check for duplicates in the dataset using the '.duplicated()' method.

In [16]:
df[df.duplicated(keep=False)]

age   sex    bmi  children smoker     region    charges
195   19  male  30.59         0     no  northwest  1639.5631
581   19  male  30.59         0     no  northwest  1639.5631

* The above shows that the two rows (195 and 581) contain duplicate values. we can use the '.drop_duplicates()' method to drop one of the duplicates leaving the first instance of the values.

In [20]:
df.drop_duplicates(keep='first', inplace=True)
df.duplicated().sum()

np.int64(0)

---

# Section 2

Section 2 content

---

NOTE

* You may add as many sections as you want, as long as it supports your project workflow.
* All notebook's cells should be run top-down (you can't create a dynamic wherein a given point you need to go back to a previous cell to execute some task, like go back to a previous cell and refresh a variable content)

---

# Push files to Repo

* In cases where you don't need to push files to Repo, you may replace this section with "Conclusions and Next Steps" and state your conclusions and next steps.

In [21]:
import os
try:
  # create your folder here
  # os.makedirs(name='')
except Exception as e:
  print(e)


IndentationError: expected an indented block after 'try' statement on line 2 (553063055.py, line 5)